src/components/tools/text-tool.tsx

```diff
@@ -69,11 +58,39 @@ export default class TextToolComponent extends BaseComponent<IProps, IState> {
     if (content.type === "Text") {
       if (!readOnly) {
         content.setSlate(change.value);
+
+        // The state will have updated in response to the content change, but the content loses the cursor position
+        // So, we update here using the raw change, which still contains the cursor position
+        this.setState({ value: change.value });
       }
-      this.setState({ value: change.value });
     }
   }
 
+  public componentDidMount() {
+    const inititialTextContent = this.props.model.content as TextContentModelType;
```

Comment:
```suggestion
    const initialTextContent = this.props.model.content as TextContentModelType;
```
`inititialTextContent` → `initialTextContent`

Reply:
Fixed.